[terra-arrange][terra-toggle-section-header] - Added terra-arrange Accessibility guidance and added minor changes in toggle section header (#3874)
cerner/terra-core: packages/terra-core-docs/src/terra-dev-site/doc/arrange/AccessibilityGuide.2.doc.mdx

> # Accessibility Guide for Terra Arrange

Reviewer: Add the badges 
```suggestion
import { Badge } from 'terra-arrange/package.json?dev-site-package';

<Badge />

# Accessibility Guide for Terra Arrange
```

Author: Added.

> ## Why the accessibility of Arrange is important ?

Reviewer 2: Followed as per the accessibility guide format

> ## Why the accessibility of Arrange is important ?

Reviewer 3: What is the format that you're referring to? The original sentence has grammar errors and does not read correctly.

> ## Why the accessibility of Arrange is important ?

Reviewer 3: cc: @eawww

> ## Why the accessibility of Arrange is important ?

Reviewer 2: This can be changed to 'Why is it important for Terra Arrange to meet accessibility guidelines?'

> ## Why the accessibility of Arrange is important ?

Author: Updated

> > Arrange is a component which is used for performing the layout alignment in the position of top/bottom/right/left. Though the CSS alignment does not affect the accessibility, the reading order of the content need to be maintained same as the visual order. If the content is aligned with any position property to make it look in different place which will create a complexity for the screen reader user to get the correct reading order.   

Reviewer 2: We can do this minor change

> > Arrange is a component which is used for performing the layout alignment in the position of top/bottom/right/left. Though the CSS alignment does not affect the accessibility, the reading order of the content need to be maintained same as the visual order. If the content is aligned with any position property to make it look in different place which will create a complexity for the screen reader user to get the correct reading order.   

Reviewer 2: The line can be changed to 'CSS alignment does not affect the accessibility, however it is crucial to ensure that the screen reader reads the content in the same manner as to how the content is visually present.'

> > Arrange is a component which is used for performing the layout alignment in the position of top/bottom/right/left. Though the CSS alignment does not affect the accessibility, the reading order of the content need to be maintained same as the visual order. If the content is aligned with any position property to make it look in different place which will create a complexity for the screen reader user to get the correct reading order.   

Author: Updated.

> > Arrange is a component which is used for performing the layout alignment in the position of top/bottom/right/left. Though the CSS alignment does not affect the accessibility, the reading order of the content need to be maintained same as the visual order. If the content is aligned with any position property to make it look in different place which will create a complexity for the screen reader user to get the correct reading order.   

Author: Updated.

>
> Making the DOM order same as visual order will help the assistive technology users to get the correct reading order. 
>

> > Arrange may also have the interactive elements which needs to be keyboard focusable and acted upon keyboard Enter/Space key. 

Reviewer 3: ```suggestion
> Arrange may also have interactive elements which need to be keyboard focusable and interactable using the `Enter` or `Space` keys. 
```

Reviewer 2: We can do this minor change

Author: Updated.

>
>&nbsp;&nbsp;

> - **Blind screen reader users** do not necessarily interact with the page in a linear fashion and can use our component out of context. Hence, we need to provide descriptive label for each interactive element. 

Reviewer 3: Not all screen reader users are blind :)

```suggestion
- **Screen reader users** do not necessarily interact with the page in a linear fashion and can use Arrange out of context. Hence, a descriptive label is necessary for each interactive element. 
```

Reviewer 2: We can do this minor change

Author: Updated.

  - Terra components do not cause a change of context on focus by default.  
  - Engineers must ensure that no unexpected change of context happens when interactive elements receive focus or when focus is removed (on blur).  
  - Content creators must inform engineers that it should not cause any change while on focus.   

- [3.2.2 On Input](https://wiki.cerner.com/display/UserExperience/3.2.2+On+Input) — Supports 
  - Terra Arrange does not cause an unexpected change of context on input. 
  - Engineers must ensure nothing unexpected happens on input.  
  - Content creators must also ensure buttons/icons do not create a change of context for the user. If there is a change of context on input, the user must be warned (with text) before the user activates the button.   

### Supported Features and Technology  
- Keyboard Interactions 
- JAWS Support with Chrome (PC) 
- VoiceOver Support with Chrome, Safari (MAC) 

### Partial Support & Requiring Further Enhancement  
- NVDA not tested (PC) 
- Mobile Touch Interactions with Screen Reader assistive technology 
- Speech Input Interactions with assistive technology 

> - [Report a problem with this component on GitHub](https://github.com/cerner/terra-framework/issues/new/choose) 

Reviewer: This 
```suggestion
- [Report a problem with this component on GitHub](https://github.com/cerner/terra-core/issues/new/choose) 
```

Author: Updated
